src/tracing/trace_config_parser.cc

```diff
@@ -0,0 +1,116 @@
+#include "tracing/trace_config_parser.h"
+
+#include <string.h>
+
+namespace node {
+namespace tracing {
+
+using v8::Array;
+using v8::Boolean;
+using v8::HandleScope;
+using v8::JSON;
+using v8::NewStringType;
+using v8::String;
+
+// String options that can be used to initialize TraceOptions.
+static const char* kRecordUntilFull = "record-until-full";
+static const char* kRecordContinuously = "record-continuously";
+static const char* kRecordAsMuchAsPossible = "record-as-much-as-possible";
+
+static const char* kRecordModeParam = "record_mode";
+static const char* kEnableSamplingParam = "enable_sampling";
+static const char* kEnableSystraceParam = "enable_systrace";
+static const char* kEnableArgumentFilterParam = "enable_argument_filter";
+static const char* kIncludedCategoriesParam = "included_categories";
+static const char* kExcludedCategoriesParam = "excluded_categories";
+
+void TraceConfigParser::FillTraceConfig(
+    Isolate* isolate, TraceConfig* trace_config, const char* json_str) {
+  HandleScope handle_scope(isolate);
+  Local<Context> context = Context::New(isolate);
+  Context::Scope context_scope(context);
+
+  Local<String> source =
+      String::NewFromUtf8(isolate, json_str, NewStringType::kNormal)
+          .ToLocalChecked();
+  Local<Value> result = JSON::Parse(context, source).ToLocalChecked();
+  Local<Object> trace_config_object = Local<Object>::Cast(result);
+
+  trace_config->SetTraceRecordMode(
+      GetTraceRecordMode(isolate, context, trace_config_object));
+  if (GetBoolean(isolate, context, trace_config_object, kEnableSamplingParam))
+    trace_config->EnableSampling();
+  if (GetBoolean(isolate, context, trace_config_object, kEnableSystraceParam))
+    trace_config->EnableSystrace();
+  if (GetBoolean(
+      isolate, context, trace_config_object, kEnableArgumentFilterParam))
+    trace_config->EnableArgumentFilter();
+  UpdateCategoriesList(isolate, context, trace_config_object,
+                       kIncludedCategoriesParam, trace_config);
+  UpdateCategoriesList(isolate, context, trace_config_object,
+                       kExcludedCategoriesParam, trace_config);
+}
+
+bool TraceConfigParser::GetBoolean(Isolate* isolate, Local<Context> context,
+                                   Local<Object> object, const char* property) {
+  Local<Value> value = GetValue(isolate, context, object, property);
+  if (value->IsNumber()) {
```

Comment:
Should this be:
```javascript
if (value->IsNumber() || value->IsBoolean())
```
I'm not certain `IsNumber()` is false for a boolean literal, but I'm assuming that's the case.